be/test/vec/columns/column_array_test.cpp

```diff
@@ -0,0 +1,425 @@
+}
+
+TEST_F(ColumnArrayTest, FieldTest) {
+    assert_field_callback(array_columns, serdes);
+}
+
+TEST_F(ColumnArrayTest, GetRawDataTest) {
+    EXPECT_ANY_THROW({ array_columns[0]->get_raw_data(); });
+}
+
+TEST_F(ColumnArrayTest, GetBoolTest) {
+    EXPECT_ANY_THROW({ array_columns[0]->get_bool(0); });
+}
+
+TEST_F(ColumnArrayTest, GetIntTest) {
+    EXPECT_ANY_THROW({ array_columns[0]->get_int(0); });
+}
+
+TEST_F(ColumnArrayTest, SerDeVecTest) {
+    ser_deser_vec(array_columns, array_types);
+}
+
+TEST_F(ColumnArrayTest, serDeserializeWithArenaImpl) {
+    ser_deserialize_with_arena_impl(array_columns, array_types);
+}
+
+TEST_F(ColumnArrayTest, SizeTest) {
+    assert_size_callback(array_columns, serdes);
+}
+
+TEST_F(ColumnArrayTest, ByteSizeTest) {
+    assert_byte_size_callback(array_columns, serdes);
+}
+
+TEST_F(ColumnArrayTest, AllocateBytesTest) {
+    assert_allocated_bytes_callback(array_columns, serdes);
+}
+
+TEST_F(ColumnArrayTest, PopbackTest) {
+    assert_pop_back_callback(array_columns, serdes);
+}
+
+TEST_F(ColumnArrayTest, CloneTest) {
+    assert_clone_resized_callback(array_columns, serdes);
+}
+
+TEST_F(ColumnArrayTest, CutTest) {
+    assert_cut_callback(array_columns, serdes);
+}
+
+TEST_F(ColumnArrayTest, ResizeTest) {
+    assert_resize_callback(array_columns, serdes);
+}
+
+TEST_F(ColumnArrayTest, ReserveTest) {
+    assert_reserve_callback(array_columns, serdes);
+}
+
+TEST_F(ColumnArrayTest, ReplicateTest) {
+    assert_replicate_callback(array_columns, serdes);
+}
+
+TEST_F(ColumnArrayTest, ReplaceColumnTest) {
+    assert_replace_column_data_callback(array_columns, serdes);
+    assert_replace_column_null_data_callback(array_columns, serdes);
+}
+
+TEST_F(ColumnArrayTest, AppendDataBySelectorTest) {
+    assert_append_data_by_selector_callback(array_columns, serdes);
+}
+
+TEST_F(ColumnArrayTest, PermutationAndSortTest) {
+    for (int i = 0; i < array_columns.size(); i++) {
+        auto& column = array_columns[i];
+        auto& type = array_types[i];
+        auto column_type = type->get_name();
+        std::cout << "column_type: " << column_type << std::endl;
+        // permutation
+        assert_column_permutations(column->assume_mutable_ref(), type);
+    }
+}
+
+TEST_F(ColumnArrayTest, FilterTest) {
+    assert_filter_callback(array_columns, serdes);
+}
+
+// HASH Interfaces
+TEST_F(ColumnArrayTest, HashTest) {
+    // XXHash
+    assert_update_hashes_with_value_callback(array_columns, serdes);
+    // XXhash with null_data
+
+    // CrcHash
+    std::vector<PrimitiveType> pts(array_columns.size(), PrimitiveType::TYPE_ARRAY);
+    assert_update_crc_hashes_callback(array_columns, serdes, pts);
+    // CrcHash with null_data
+};
+
+//////////////////////// special function from column_array.h ////////////////////////
+TEST_F(ColumnArrayTest, CreateArrayTest) {
+    // test create_array : nested_column && offsets_column should not be const, and convert_to_full_column_if_const should not impl in array
```

Comment:
Why?

Reply:
because for convert_to_full_column_if_const  semantics in IColumn:
```
    /** If column isn't constant, returns nullptr (or itself).
      * If column is constant, transforms constant to full column (if column type allows such transform) and return it.
      */
``` now this implemention for array is not right to create a new column , we should return itself .
then this implemention  will make current array_column with nested data_column use_count not equal which make other logic error , to avoid this, we should delete this implemention
but how if we make const_column in array, then we call IColumn::convert_to_full_column_if_const , we just return itself without do nothing which is not right result , so we can just call convert_to_full_column_if_const first in create action